modules/oe_translation_cdt/modules/oe_translation_cdt_mock/src/Plugin/ServiceMock/StatusApi.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_translation_cdt_mock\Plugin\ServiceMock;
+
+use Drupal\Core\Site\Settings;
+use Drupal\oe_translation_cdt\Mapper\LanguageCodeMapper;
+use Drupal\oe_translation_remote\TranslationRequestRemoteInterface;
+use GuzzleHttp\Psr7\Response;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Intercepts status request made to CDT.
+ *
+ * @ServiceMock(
+ *   id = "oe_translation_cdt_status_api",
+ *   label = @Translation("CDT mocked status responses for testing."),
+ *   weight = 0,
+ * )
+ */
+class StatusApi extends ServiceMockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEndpointUrl(): string {
+    return Settings::get('cdt.status_api_endpoint');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getEndpointResponse(RequestInterface $request): ResponseInterface {
+    // The entity uses the same identifier as the mocked CDT request.
+    $parameters = $this->getRequestParameters($request);
+    $response = json_decode($this->getResponseFromFile('status_response_200.json'), TRUE);
+    /** @var \Drupal\oe_translation_cdt\TranslationRequestCdtInterface|null $entity */
+    $entity = $this->entityTypeManager->getStorage('oe_translation_request')->load($parameters['requestnumber']);
+    if (!$entity) {
+      $this->log('400: Failed to get the mock status. The translation request does not exist.', $request);
+      return new Response(400, [], $this->getResponseFromFile('status_response_400.json'));
+    }
+
+    // Change only the important parameters.
```

Comment:
I did not change any other parameters of this request, as Drupal does not check it for now. This may change in [D8AGE-291](https://citnet.tech.ec.europa.eu/CITnet/jira/browse/D8AGE-291), I will modify the mock accordingly.